stdpopsim/catalog/HomSap/__init__.py

```diff
@@ -1650,3 +1650,175 @@ def _AJ():
+    ]
+    citations = [
+        stdpopsim.Citation(
+            author="Jouganous et al.",
+            year="2017",
+            doi="https://doi.org/10.1534/genetics.117.200493",
+            reasons={stdpopsim.CiteReason.DEM_MODEL},
+        )
+    ]
+
+    generation_time = 29
+
+    # Parameter values from Table 4 (in bold)
+
+    # Population sizes
+    N_A = 11293  # ancestral YRI population size
+    N_AF = 23721  # YRI population size
+    N_B = 2831  # B population size (out-of-Africa)
+    N_EU0 = 2512  # CEU population size (intial)
+    N_AS0 = 1019  # Asian population size (inital)
+    N_JP0 = 4384  # Japanese population size (inital)
+
+    # growth rates
+    r_EU = 0.0016  # European growth rate
+    r_JP = 0.0129  # Japanese growth rate
+    r_AS = 0.0026  # Asian growth rate
+
+    # migration rates
+    m_AF_B = 16.8e-5  # YRI <--> B (out-of-Africa)
+    m_AF_EU = 1.14e-5  # YRI <--> CEU
+    m_AF_AS = 0.56e-5  # YRI <--> CHB
+    m_EU_AS = 4.75e-5  # European <--> CHB
+    m_CH_JP = 3.3e-5  # CHBHan <--> JPT
+
+    # epochs (specified in years, converted to generations)
+    T_AF = int(357000 / generation_time)  # time ancestral to African
+    T_B = int(119000 / generation_time)  # time of Out of Africa split
+    T_EU_AS = int(46000 / generation_time)  # time of Europe/Asia split
+    T_CH_JP = int(9000 / generation_time)  # time of Han/Japanese split
+
+    # also
+    r_AF = 0  # YRI and B growth rate
+
+    # A and YRI
+    pop0 = msprime.PopulationConfiguration(
+        initial_size=N_AF * math.exp(r_AF * T_AF),
+        growth_rate=r_AF,
+        metadata=populations[0].asdict(),
+    )
+
+    # B and CEU
+    pop1 = msprime.PopulationConfiguration(
+        initial_size=N_EU0 * math.exp(r_EU * T_EU_AS),
+        growth_rate=r_EU,
+        metadata=populations[1].asdict(),
+    )
+
+    # Asian and CHB
+    pop2 = msprime.PopulationConfiguration(
+        initial_size=N_AS0 * math.exp(r_AS * T_EU_AS),
+        growth_rate=r_AS,
+        metadata=populations[2].asdict(),
+    )
+
+    # JPT
+    pop3 = msprime.PopulationConfiguration(
+        initial_size=N_JP0 * math.exp(r_JP * T_CH_JP),
+        growth_rate=r_JP,
+        metadata=populations[3].asdict(),
+    )
+
+    population_configurations = [pop0, pop1, pop2, pop3]
+
+    migration_matrix = [
+        [0, m_AF_EU, m_AF_AS, 0],
+        [m_AF_EU, 0, m_EU_AS, 0],
+        [m_AF_AS, m_EU_AS, 0, m_CH_JP],
+        [0, 0, m_CH_JP, 0],
+    ]
+
+    # Demographic events from most recent -> oldest
+
+    # JPT splits from CHB
+    JPT_event = [
+        # seg migration rate JPT <--> CHB to zero
+        msprime.MigrationRateChange(time=T_CH_JP, rate=0.0, matrix_index=(2, 3)),
+        msprime.MigrationRateChange(time=T_CH_JP, rate=0.0, matrix_index=(3, 2)),
+        # JPT splits from CHB
+        msprime.MassMigration(time=T_CH_JP, source=3, destination=2, proportion=1.0),
+        # set growth rate in JPT to zero
+        msprime.PopulationParametersChange(
```

Comment:
This parameters change shouldn't be necessary, as no lineages remain in the JPT population.

Reply:
Yes, that makes sense.  I was struggling a bit how to deal with pops before they should exist.